include/deal.II/base/subscriptor.h

```diff
@@ -236,6 +250,24 @@ Subscriptor::serialize(Archive &,
   // documentation of this function
 }
 
+template <typename StreamType>
+inline
+void
+Subscriptor::list_subscribers(StreamType &stream) const
+{
+#ifdef DEAL_II_WITH_THREADS
+  std::lock_guard<std::mutex> lock(mutex);
+#endif
+
+  for (map_iterator it = counter_map.begin();
+       it != counter_map.end(); ++it)
+    stream
+        << it->second << '/'
+        << counter << " subscriptions from \""
+        << it->first << '\"'
+        << std::endl;
+}
```

Comment:
I don't think that using a lock is necessary here. I believe that it should be up to the caller to not print to object from multiple threads simultaneously.

Reply:
I think we do. We still want to make sure that the map doesn't change when running through all elements of the map.
We could use a `shared_lock` here as well.

Reply:
Thats a good point. We should keep the lock for this reason.

Reply:
> We could use a shared_lock here as well.

@masterleinad Apparently this is a [c++14 feature](http://en.cppreference.com/w/cpp/thread/shared_lock), so we'll have to do with the mutex + lock guard option.

Reply:
I know, but we could `boost::shared_lock`. However, I am not sure if this is worth it.

Reply:
Ah right, I forgot that such functionality would be covered by boost. Thanks :-)